File: 53-maximum-subarray/53-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxsum=Integer.MIN_VALUE;
        int sum=0,i=0,j=0;
        while(j<nums.length){
            sum=sum+nums[j];
            if(maxsum<sum) maxsum=sum;
            j++;
            if(sum<0) {
                i=j;
                sum=0;
            }
        }
        return maxsum;
    }
}

File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sum=0,maxsum=INT_MIN;
        int i=0;
        for(int j=0;j<nums.size();j++){
            sum+=nums[j];
            if(sum>maxsum){
                maxsum=sum;
            }
            if(sum<0){
                sum=0;
                i=j+1;
            }
        }
        return maxsum;
    }
};